### PR TITLE
 Tasks 24/32: FE: Aside filter component. Implement sort widge/Implement product list pagination component

### PR DESCRIPTION
**Pull request by di-kinosian**

### CODE REVIEW

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  totalPages?: number;
+  onPageChange: (page: number) => void;
+}
+
+export const PaginationController = ({
+  page,
+  onPageChange,
+  totalPages,
+}: Props) => {
+  const [currentPage, setCurrentPage] = React.useState<number>(page);
```
Comment (VYaniv) [inline src/components/Pagination/Pagination.tsx:18]: Direct import please

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  const classes = useStyles();
```
Comment (VYaniv) [inline src/components/Pagination/Pagination.tsx:20]: Unneeded empty line

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  const handleChange = (event: React.ChangeEvent<unknown>, value: number) => {
```
Comment (VYaniv) [inline src/components/Pagination/Pagination.tsx:22]: Returned type

File: src/components/Select/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { Options } from "./types";
+
+interface Props {
+  width?: number;
+  options: Options[];
+  getParams?: (value: string) => void;
+}
+
+export const Select = ({ width, options, getParams }: Props) => {
+  const [selectValue, setSelectValue] = useState("");
```
Comment (VYaniv) [inline src/components/Select/Select.tsx:17]: state type

File: src/pages/Products/Products.tsx [hunk 5]
```diff
@@ -1,44 +1,124 @@
...
 
 export const Products = () => {
   const dispatch: AppDispatch = useDispatch();
-  const { products } = useAppSelector((state) => state.product);
   const { selectedFilters } = useAppSelector((state) => state.filters);
+  const { products, total, limit, page } = useAppSelector(
+    (state) => state.product,
+  );
+
+  const [sortData, setSortData] = useState<{ field: string; order: string }>();
```
Comment (VYaniv) [inline src/pages/Products/Products.tsx:24]: Unneeded empty line

File: src/pages/Products/Products.tsx [hunk 6]
```diff
@@ -1,44 +1,124 @@
...
     dispatch(
       getAllProducts({
-        page: 1,
+        page: currentPage,
         limit: 5,
         brands: selectedFilters.brands,
         priceMax: selectedFilters.price.max,
         priceMin: selectedFilters.price.min,
         rating: selectedFilters.rating || 5,
+        sortField: sortData?.field,
```
Comment (VYaniv) [inline src/pages/Products/Products.tsx:45]: Limit can be a constant

File: src/pages/Products/Products.tsx [hunk 7]
```diff
@@ -1,44 +1,124 @@
...
       }),
     );
-  }, [selectedFilters]);
+  }, [selectedFilters, sortData, currentPage]);
+
+  useEffect(() => {
+    if (showMoreMode) {
+      setVisibleProducts((prev) => {
+        const newProducts = products.filter(
+          (product) => !prev.some((p) => p._id === product._id),
```
Comment (VYaniv) [inline src/pages/Products/Products.tsx:55]: `({id}) => _id ===` to avoid `p`

Comment (VYaniv) [inline src/pages/Products/Products.tsx:55]: Why do you need filtering here?

File: src/pages/Products/Products.tsx [hunk 8]
```diff
@@ -1,44 +1,124 @@
...
+    }
+  }, [products, showMoreMode]);
+
+  const handlePageChange = (newPage: number): void => {
+    setCurrentPage(newPage);
+    setShowMoreMode(false);
+    setVisibleProducts([]);
+  };
+
+  const handleShowMore = () => {
```
Comment (VYaniv) [inline src/pages/Products/Products.tsx:70]: Returned type

File: src/components/Pagination/Pagination.tsx (hunk 1 above)
Comment (di-kinosian) [reply src/components/Pagination/Pagination.tsx]: done

File: src/components/Pagination/Pagination.tsx (hunk 2 above)
Comment (di-kinosian) [reply src/components/Pagination/Pagination.tsx:18]: done

File: src/components/Pagination/Pagination.tsx (hunk 3 above)
Comment (di-kinosian) [reply src/components/Pagination/Pagination.tsx]: done

File: src/components/Select/Select.tsx (hunk 4 above)
Comment (di-kinosian) [reply src/components/Select/Select.tsx]: done

File: src/pages/Products/Products.tsx (hunk 5 above)
Comment (di-kinosian) [reply src/pages/Products/Products.tsx:25]: done

File: src/pages/Products/Products.tsx (hunk 6 above)
Comment (di-kinosian) [reply src/pages/Products/Products.tsx]: done

File: src/pages/Products/Products.tsx (hunk 7 above)
Comment (di-kinosian) [reply src/pages/Products/Products.tsx]: the same

File: src/pages/Products/Products.tsx (hunk 8 above)
Comment (di-kinosian) [reply src/pages/Products/Products.tsx]: done

File: src/pages/Products/Products.tsx [hunk 9]
```diff
@@ -1,44 +1,123 @@
...
+  }, [selectedFilters, sortData, currentPage]);
+
+  useEffect(() => {
+    if (showMoreMode) {
+      setVisibleProducts((prev) => {
+        const uniqueProducts = new Map();
+        [...prev, ...products].forEach((product) => {
+          uniqueProducts.set(product._id, product);
+        });
+        return Array.from(uniqueProducts.values());
```
Comment (VYaniv) [inline src/pages/Products/Products.tsx:57]: Empty line before return. But I have a couple of questions about this part. Let`s speak tomorrow

**Review by VYaniv — APPROVED** [review]

**Pull request merged**